client/src/js/directives/bhLocationSelect.js

```diff
@@ -77,32 +78,29 @@ function LocationSelectController(Locations, $scope) {
 
   /**
    * <select> component messages to be translated
-   * if there is no data, indicate that to the user.
    */
-  var selectCountry  = 'SELECT.COUNTRY';
-  var selectProvince = 'SELECT.PROVINCE';
-  var selectSector   = 'SELECT.SECTOR';
-  var selectVillage  = 'SELECT.VILLAGE';
-  var noData         = 'SELECT.EMPTY';
   vm.messages = {
-    country:  selectCountry,
-    province: selectVillage,
-    sector:   selectSector,
-    village:  selectVillage,
+    country:  Locations.messages.country,
+    province: Locations.messages.province,
+    sector:   Locations.messages.sector,
+    village:  Locations.messages.village,
   };
 
-  // load the countries once, at startup
-  Locations.countries()
-  .then(function (countries) {
-    vm.countries = countries;
-
-    // if there are countries to select, show a "select a country" message
-    // however, if there isn't any data, show a "no data" message. This pattern
-    // is used throughout the component.
-    vm.messages.country = (countries.length > 0) ?
-      selectCountry :
-      noData;
-  });
+  function loadCountries() {
+    Locations.countries()
```

Comment:
Thanks for the feedback.

I'm happy to change the `LocationService` API.  However, since this is an API change, it will break the `bhLocationSelect` component and any other modules/services that use the `LocationService` (I'm not aware of any).

Furthermore, this change is not for code introduced in this PR (the API was developed and introduced in [this commit](https://github.com/IMA-WorldHealth/bhima-2.X/commit/55932e8b14a50d0af807d4b15ffd6541073d9a9d) two weeks ago), I propose that a separate pull request change the service API, update the `bhLocationSelect` component and any tests that are needing to be updated should be addressed then.  If we try to include all these changes in this PR, the original purpose will be lost.